src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -21,7 +22,9 @@ public class ContentResult : ActionResult
 
             if (!string.IsNullOrEmpty(ContentType))
             {
-                response.ContentType = ContentType;
+                var contentTypeHeader = new MediaTypeHeaderValue(ContentType);
+                contentTypeHeader.Encoding = ContentEncoding;
+                response.ContentType = contentTypeHeader.ToString();
             }
 
             if (Content != null)
```

Comment:
if Content is null we do not write the response, so should we return `Task.FromResult(true)` in this case... @Tratcher : will this be a non-complete task? and if yes, can it cause problems in hosts?...like my thought was that headers are sent the moment body is being written, but here we just set the headers and we do not write any content...so how would the host know to send the headers anyway...does it look for completed tasks?